Add collaboration workflow document (#2507)
posit-dev/publisher: extensions/vscode/README.md

[Updating Previously Deployed Content](https://github.com/posit-dev/publisher/blob/main/docs/vscode.md#updating-previously-deployed-content)
documentation for more information.

> ### Seemless collaboration

Reviewer: ```suggestion
### Seamless collaboration
```

Author: Thanks for catching this - fixed in: https://github.com/posit-dev/publisher/pull/2507/commits/e1ce73d49e28ac0fc3418e9dcccbdca622da0665

Collaborate with your team using shared configurations and deployments. Easily
pick up where others left off with deployed content.

> See the [Collaboration Workflow](https://github.com/posit-dev/publisher/blob/main/docs/collaboration.md)
> documentation for more information.

Reviewer: Minor, I might suggest using the title of the document as the link text, since it's a good explanation of what they'll read when they get there (and the page does not describe an individual workflow per se, but has options depending on how various systems are set up):

```suggestion
See [Collaborative Publishing with Posit Publisher](https://github.com/posit-dev/publisher/blob/main/docs/collaboration.md) for more information.
```

[Updating Previously Deployed Content](https://github.com/posit-dev/publisher/blob/main/docs/vscode.md#updating-previously-deployed-content)
documentation for more information.

### Seemless collaboration

Collaborate easily with your team by sharing your configurations and
deployments. Others can pick up where you left off and

> Allow collaborators to easily pick up where you left off by using the `.posit`

Reviewer 2: Should we switch this to call out the `.posit/publisher` files rather than just the `.posit` files?

Author: Yes that is a great point. I'll get that updated.
